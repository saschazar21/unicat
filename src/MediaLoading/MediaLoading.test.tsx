import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Loader from './MediaLoading';

describe('Loader', () => {
  it('renders', () => {
    const wrapper = shallow(<Loader />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.dark')).toHaveLength(0);
  });

  it('renders a dark variant', () => {
    const wrapper = shallow(<Loader dark />);

    expect(wrapper.find('.dark')).toHaveLength(1);
  });
});
