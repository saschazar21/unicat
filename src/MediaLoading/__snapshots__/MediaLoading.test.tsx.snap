// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders 1`] = `
<svg
  className="wrapper"
  viewBox="0 0 512 384"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <linearGradient
      id="Loader-7820-000"
    >
      <stop
        className="outer"
        offset="0%"
      />
      <stop
        className="inner"
        offset="50%"
      />
      <stop
        className="outer"
        offset="100%"
      />
      <animate
        attributeName="x2"
        dur="4s"
        fill="freeze"
        from="-0.5"
        repeatCount="indefinite"
        to="1.5"
      />
      <animate
        attributeName="x1"
        dur="4s"
        fill="freeze"
        from="-1.0"
        repeatCount="indefinite"
        to="1"
      />
    </linearGradient>
  </defs>
  <g
    fill="url(#Loader-7820-000)"
    fillRule="evenodd"
  >
    <path
      d="M512 0v384H0V0h512zm-8 8H8v368h496V8z"
      fillRule="nonzero"
    />
    <path
      d="M488 24v304H24V24h464zm-8 8H32v288h448V32z"
      fillRule="nonzero"
    />
    <path
      d="M144 176l40 64h-16l32 32h-21.714L224 304H64l45.714-32H88l32-32h-16l40-64zm0 15.094L118.434 232h20.88l-32 32h27.779l-45.714 32H198.62l-45.714-32h27.78l-32-32h20.879L144 191.094z"
      fillRule="nonzero"
    />
    <path
      d="M276 154l42 67h-16.8l33.6 33.5H312l48 33.5H192l48-33.5h-22.8l33.6-33.5H234l42-67zm-27.543 59h21.697l-33.6 33.5h28.887l-48 33.5h117.118l-48-33.5h28.887l-33.6-33.5h21.697L276 169.062 248.457 213z"
      fillRule="nonzero"
    />
    <path
      d="M34.656 194.991l52.319-46.447 46.182 15.944 60.052-46.648 15.319 23.799L256 108.861l91.68 63.303 62.175-8.049 68.472 31.519 3.346-7.268-70.563-32.481-61.411 7.951L256 99.139l-45.218 31.222-15.577-24.201-63.534 49.352-46.508-16.056-55.819 49.553zM384 56c22.091 0 40 17.909 40 40s-17.909 40-40 40-40-17.909-40-40 17.909-40 40-40zm0 8c-17.673 0-32 14.327-32 32 0 17.673 14.327 32 32 32 17.673 0 32-14.327 32-32 0-17.673-14.327-32-32-32z"
      fillRule="nonzero"
    />
  </g>
</svg>
`;
